packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3536,11 +3552,81 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  final bool _platformSupportsFadeOnScroll = switch (defaultTargetPlatform) {
+    TargetPlatform.android => true,
+    TargetPlatform.iOS => true,
+    TargetPlatform.fuchsia => false,
+    TargetPlatform.linux => false,
+    TargetPlatform.macOS => false,
+    TargetPlatform.windows => false,
+  };
+
   void _onEditableScroll() {
-    _selectionOverlay?.updateForScroll();
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      _selectionOverlay?.updateForScroll();
+    }
     _scribbleCacheKey = null;
   }
 
+  void _handleContextMenuOnParentScroll(ScrollNotification notification) {
+    if (_internalScrolling) {
+      return;
+    }
+    _handleContextMenuOnScroll(notification);
+  }
+
+  Rect _calculateViewportRect() {
```

Comment:
If this is the scrollable RenderViewport, I think you could get the rect from RenderAbstractViewport.of(context)

Reply:
My intention here was to get a rect that closest represents the visible portion of the flutter view. I want to know if the `RenderEditable` is visible on the screen. Would `RenderAbstractViewport.of(context)` still work if we have a TextField within a horizontal scrollable nested within a vertical scrollable?

Reply:
In that case you may have to bubble up through multiple viewports to find out. Maybe calling getOffsetToReveal for each viewport will work?

Check this out from showInViewport, this is the case where we determine, this is visible we don't need to make it visible. I don't think you need to do the things in the if clause here, it's jsut an example of how we determined it was visible:

https://github.com/flutter/flutter/blob/023e5addaa6e8e294a200cf754afaa1656f14aa6/packages/flutter/lib/src/rendering/viewport.dart#L1248-L1250

Reply:
Thanks for the suggestion! Something like this works for me, but I realized `RenderAbstractViewport` doesn't take into account things like status bar or keyboard. In this case maybe the current implementation of calculating the approximate visible viewport rect might be more accurate, what do you think? 

```dart
  bool? _renderEditableIsVisible(ScrollNotification notification) {
    RenderAbstractViewport? closestViewport = RenderAbstractViewport.maybeOf(renderEditable);
    if (closestViewport == null) {
      return null;
    }
    while (closestViewport != null) {
      final RevealedOffset leadingEdgeOffset = closestViewport.getOffsetToReveal(renderEditable, 0.0);
      final RevealedOffset trailingEdgeOffset = closestViewport.getOffsetToReveal(renderEditable, 1.0);
      final double currentOffset = -notification.metrics.pixels;
      final RevealedOffset? targetOffset = RevealedOffset.clampOffset(
        leadingEdgeOffset: leadingEdgeOffset,
        trailingEdgeOffset: trailingEdgeOffset,
        currentOffset: currentOffset,
      );
      if (targetOffset != null) {
        // `descendant` is between leading and trailing edge and hence already
        //  fully shown on screen. No action necessary.
        return false;
      }
      closestViewport = RenderAbstractViewport.maybeOf(closestViewport.parent);
    }
    return true;
  }
```

Reply:
SGTM! 🎉